pages/api/login.js

```diff
@@ -0,0 +1,35 @@
+/* eslint-disable camelcase */
+import { withIronSessionApiRoute } from 'iron-session/next';
+
+import httpClient from 'lib/httpClient';
+import { sessionOptions } from 'lib/session';
+
+export default withIronSessionApiRoute(loginRoute, sessionOptions);
+
+async function loginRoute(req, res) {
```

Comment:
Could you point me out why do we need to add another level of API to call the API? 💭 
IMO having some adapters to fetch the data from the API on the client side is quite enough 🤔

Reply:
I actually wanted to use static generation, here is what I replied to Xavier's comment in https://github.com/tyrro/react-survey/pull/54#discussion_r842549872

In a word, Static generation with server side loading state, and then fetching data in the clients - it also hides the server information from the client.

Reply:
UPDATE:
Due to two levels of API as pointed out, I decided to drop nextjs API endpoints in favor of client side data fetching!
Thanks @carryall !